bgpd/bgp_route.c

```diff
@@ -10247,6 +10247,22 @@ bgp_distance_apply (struct prefix *p, struct bgp_info *rinfo, afi_t afi,
     }
 }
 
+static void
+announce_routes_distance_update (struct bgp *bgp)
```

Comment:
Let's call this function bgp_announce_routes_distance_update.

Second, pass in the afi/safi of the bgp distance command and only process that particular safi.  If I change ipv4/unicast why would I reprocess ipv6/unicast( as an example)?